ui/src/app/dashboard/dashboard.resolver.service.ts

```diff
@@ -0,0 +1,15 @@
+import {Injectable} from "@angular/core";
+import {ActivatedRouteSnapshot, Resolve, RouterStateSnapshot} from "@angular/router";
+import {AggregationResponse} from "../shared/model/AggregationResponse";
+import {JobManagerService} from "../core/job-manager.service";
+import {TimeFrame} from "../shared/model/TimeFrame";
+
+@Injectable()
+export class DashboardResolver implements Resolve<AggregationResponse>{
+  constructor(
+    private jobManagerService: JobManagerService,
+  ) {}
+  resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Promise<AggregationResponse> {
+    return this.jobManagerService.queryAggregations(TimeFrame.ALLTIME, route.queryParams['projectId']);
```

Comment:
nit: I think lets start with one/two weeks for the initial implementation

Reply:
fixed.